docs/02-app/01-building-your-application/08-deploying/index.mdx

```diff
@@ -72,73 +45,187 @@ First, ensure your `package.json` has the `"build"` and `"start"` scripts:
+- Dynamically rendered pages set a `Cache-Control` header of `private, no-cache, no-store, max-age=0, must-revalidate` to prevent user-specific data from being cached. This applies to both the App Router and Pages Router. This also includes [Draft Mode](/docs/app/building-your-application/configuring/draft-mode).
 
-> **Good to know**: If a new page (with an old version) has already been prefetched by `next/link`, Next.js will use the old version. Navigating to a page that has _not_ been prefetched (and is not cached at the CDN level) will load the latest version.
+#### Static Assets
 
-## Manual Graceful shutdowns
+If you want to host static assets on a different domain or CDN, you can use the `assetPrefix` [configuration](/docs/app/api-reference/next-config-js/assetPrefix) in `next.config.js`. Next.js will use this asset prefix when retrieving JavaScript or CSS files. Separating your assets to a different domain does come with the downside of extra time spent on DNS and TLS resolution.
+
+[Learn more about `assetPrefix`](/docs/app/api-reference/next-config-js/assetPrefix).
+
+#### Configuring Caching
+
+By default, generated cache assets will be stored in memory (defaults to 50mb) and on disk. If you are hosting Next.js using a container orchestration platform like Kubernetes, each pod will have a copy of the cache. To prevent stale data from being shown since the cache is not shared between pods by default, you can configure the Next.js cache to provide a cache handler and disable in-memory caching.
+
+To configure the ISR/Data Cache location when self-hosting, you can configure a custom handler in your `next.config.js` file:
+
+```jsx filename="next.config.js"
+module.exports = {
+  experimental: {
+    incrementalCacheHandlerPath: require.resolve('./cache-handler.js'),
+    isrMemoryCacheSize: 0, // disable default in-memory caching
+  },
+}
+```
+
+Then, create `cache-handler.js` in the root of your project. This file can save the cached values anywhere, like Redis or AWS S3, for example:
+
+```jsx filename="cache-handler.js"
+const cache = new Map()
+
+module.exports = class CacheHandler {
+  constructor(options) {
+    this.options = options
+    this.cache = {}
+  }
+
+  async get(key) {
+    // This could be stored anywhere, like durable storage
+    return cache.get(key)
+  }
+
+  async set(key, data, ctx) {
+    // This could be stored anywhere, like durable storage
+    cache.set(key, {
+      value: data,
+      lastModified: Date.now(),
+      tags: ctx.tags,
+    })
+  }
+
+  async revalidateTag(tag) {
+    // Iterate over all entries in the cache
+    for (let [key, value] of cache) {
+      // If the value's tags include the specified tag, delete this entry
+      if (value.tags.includes(tag)) {
+        cache.delete(key)
+      }
+    }
+  }
+}
+```
+
+Using a custom cache handler will allow you to ensure consistency across all pods hosting your Next.js application.
```

Comment:
Hey @leerob, I noticed a lack of information about custom cache handlers in multi-instance setups. I created a one-step solution plus an example and template for Next.js. Please consider including or mentioning this example #58000 in the documentation.

I believe that the example I'm talking about could be a valuable addition to the new documentation. It provides a practical reference for users who want to implement a similar setup, which aligns with the documentation's goal of streamlining user understanding. It also ensures that users can fully utilize Next.js features in various self-hosting environments.

Reply:
Left a comment over there!